src/types/postsPrepared.ts

```diff
@@ -0,0 +1,11 @@
+import { User } from './User';
+import { Comment } from './Comment';
+
+export interface PostPrepared {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
+  user: User | null,
+  comments?: Comment[] | null,
```

Comment:
Let it be empty array if there are no comments, don't set it to null in your filter fn
And if you make it optional it means it can be undefined, but in our case it is either empty array or array of comments 
```suggestion
  comments: Comment[] ,
```